open-grant-proposals/Filecoin Financing Plan.md

```diff
@@ -0,0 +1,122 @@
+
+# Open Grant Proposal: `Filecoin Financing Plan`
+
+**Name of Project: Filecoin Financing Plan**
+
+**Proposal Category:** core-dev
+
+**Proposer:** catthehunter
+
+**Do you agree to open source all work you do on behalf of this RFP and dual-license under MIT and APACHE2 licenses?:** Yes
+
+# Project Description
+
+Please describe exactly what you are planning to build. Make sure to include the following:
+The proposal would address the issue of high threshold to participate Filecoin. For instance, the cost of the hardware and collateral for a mining pool with 1P power will be more than 3 million RMB (about $462,000). We hope that through the Filecoin financing plan, users with 
+hardware and mining pool can publish financing plans to the entire network. 
+Filecoin financing plans will include the total financing amount, the timeline, the usage plan, repayment amount (yield indicator) and repayment plan(installment). Anyone can make an investment by transferring FIL to the financing plan address, and financing repayment will be returned to the transfer account. 
+To secure the benefit of the investors, the financing plan itself shows investors the credit information of the financier. At the same time, the logic of freezing financing fund and income automatic repayment of the financing plan will be implemented in the lotus-miner, and to freeze financing fund will guarantee that the financing fund can only be used to increase the mining pool power. The Automatic income repayment will guarantee that the investors' income is not subject to the individual intentions of the financier. 
+
+## Value
+
+Please describe in more detail why this proposal is valuable for the Filecoin ecosystem. Answer the following questions:
+- What are the benefits to getting this right?
+A: The project can spread the pressure of high cost to join Filecoin, to reduce the risk for individuals to participate, to lower the cost threshold for Filecoin participation, and to attract more users to get involved.
+- What are the risks if you don't get it right?
+A: If we didn't get it right, the investors may lost their capital.
+- What are the risks that will make executing on this project difficult?
+A: The credit rating of financiers is not comprehansive or accurate enough, and that makes investors afraid to invest. 
+
+## Deliverables
+
+1. deliver the lotus-miner implementation code for financing plan
+2. deliver the web browing portal (financing plan explorer) code for financing plan and access services.
+
+## Technical Breakdown
+
+Filecoin financing plan detailed process:
+1. Add two control address types for miner, which means on the basis of the worker/owner address, we add the financing address and the repayment freeze address.
+2. A sealing sector first determines whether the financing address has some balance, If the address has, the sector will use the balance; if not, the sealing sector will follow the original logic to process.
```

Comment:
This is too closely coupled to proving which bears a big cost of increased complexity.  There is likely a better arrangement of abstractions to achieve what you want.  At the very least all of this should be handled from miner actor's withdraw method.  Ideally it would all be in a separate actor.